kernel_gateway/jupyter_websocket/__init__.py

```diff
@@ -0,0 +1,65 @@
+from ..base.handlers import default_handlers as default_base_handlers
+from ..services.activity.handlers import ActivityHandler
+from ..services.kernels.pool import KernelPool
+from ..services.kernels.handlers import default_handlers as default_kernel_handlers
+from ..services.kernelspecs.handlers import default_handlers as default_kernelspec_handlers
+from ..services.sessions.handlers import default_handlers as default_session_handlers
+from .handlers import default_handlers as default_api_handlers
+from notebook.utils import url_path_join
+
+class JupyterWebsocketPersonality(object):
+    '''Personality for standard websocket functionality, registering
+    endpoints that are part of the Jupyter Kernel Gateway API
+    '''
+    def __init__(self, kernelspec):
```

Comment:
It can be used to solve the same problem cropping up during tests, yes. Even waiting to instantiate the SessionManager until *after* calling `self.personality.init_configurables()` still leaves a race condition in the tests:
```
tornado.application: ERROR: Future <tornado.concurrent.Future object at 0x7feb145ebb00> exception was never retrieved: Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/tornado/gen.py", line 1017, in run
    yielded = self.gen.send(value)
  File "/srv/kernel_gateway/kernel_gateway/services/kernels/manager.py", line 82, in start_kernel
    if self.parent.personality.should_seed_cell(self.seed_kernelspec, code):
  File "/srv/kernel_gateway/kernel_gateway/notebook_http/__init__.py", line 84, in should_seed_cell
    return (not self.api_parser.is_api_cell(code) and not self.api_parser.is_api_response_cell(code))
AttributeError: 'NotebookHTTPPersonality' object has no attribute 'api_parser'
```